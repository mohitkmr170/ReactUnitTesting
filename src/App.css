.App {
  text-align: center;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.increment {
  background-color: blanchedalmond;
  height: 30px;
  width: 80px;
  border-radius: 8px;
}
.decrement {
  background-color: blanchedalmond;
  height: 30px;
  width: 80px;
  border-radius: 8px;
}
.textClick {
  color: blueviolet;
  font-size: 16px;
  font-family: Arial, Helvetica, sans-serif;
}
.buttonContainer {
  flex-direction: row;
}
