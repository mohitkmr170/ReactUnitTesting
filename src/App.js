import React from "react";
import logo from "./logo.svg";
import "./App.css";

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      count: 0
    };
  }

  incrementCount = amount => () => {
    this.setState(prevState => ({
      count: prevState.count + amount
    }));
  };

  increment = this.incrementCount(1);

  decrement = this.incrementCount(-1);

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <p>Count: {this.state.count}</p>
          <button className="increment" onClick={this.increment}>
            Increment
          </button>
          <button className="decrement" onClick={this.decrement}>
            Decrement
          </button>
        </header>
      </div>
    );
  }
}

export default App;
