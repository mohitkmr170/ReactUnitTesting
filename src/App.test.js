import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import App from "./App";

/*

// Basic jest test(uses enzyme) using dummy data

// Component test cases

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

// Funtional test cases

describe("addition of two numbers", () => {
  it("2+2=4", () => {
    expect(2 + 2).toBe(10);
  });
});
*/

/*

// Basic jest test(uses enzyme) using actual and expected data
// funtional test cases

describe("App Component", () => {
  it("starts with a count 0", () => {
    const wrapper = shallow(<App />);
    const text = wrapper.find("p").text();
    expect(text).toEqual("Count: 0");
  });
  it("increments count by 1 when the increment button is clicked", () => {
    const wrapper = shallow(<App />);
    const incrementBtn = wrapper.find("button.increment");
    incrementBtn.simulate("click");
    const text = wrapper.find("p").text();
    expect(text).toEqual("Count: 1");
  });
  it("decrements count by 1 when the decrement button is clicked", () => {
    const wrapper = shallow(<App />);
    const decrementBtn = wrapper.find("button.decrement");
    decrementBtn.simulate("click");
    const text = wrapper.find("p").text();
    expect(text).toEqual("Count: -1");
  });
});
*/

// Testing React components with snapshots(returns the tree(JSON format of component) and compares the changes)
// Using 'npm react-test-renderer'
// In case we intensionally changes the component, press 'U' to update the snapshot

it("matches the snapshot", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
