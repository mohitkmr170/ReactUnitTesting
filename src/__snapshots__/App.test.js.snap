// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <p>
      Count: 
      0
    </p>
    <button
      className="increment"
      onClick={[Function]}
    >
      Increment
    </button>
    <button
      className="decrement"
      onClick={[Function]}
    >
      Decrement
    </button>
  </header>
</div>
`;
